src/mongo_types.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import type { Document, ObjectId } from './bson';
+
+/** @internal */
+export type TODO_NODE_2648 = any;
+
+/** Given an object shaped type, return the type of the _id field or default to ObjectId @public */
+export type InferIdType<TSchema> = TSchema extends { _id: infer IdType } // user has defined a type for _id
+  ? {} extends IdType
+    ? Exclude<IdType, {}>
+    : unknown extends IdType
+    ? ObjectId
+    : IdType
+  : ObjectId; // user has not defined _id on schema
```

Comment:
This allows users to defined schemas that lack an `_id` property but our internal structures can still default to ObjectId when they need to reference the _id property.

Reply:
Nesting ternaries for type declarations seems a bit counter intuitive and if I were trying to use this it might be a bit confusing, what do you think about using some combination of union types, and the `Extract` and/or `Pick` utility types? Something like `Extract<Pick<TSchema, '_id'> | ObjectId, NonNullable>` (not perfect/working, just a general idea/path, sorry if I'm annoying/overstepping!)

Reply:
We'll revisit this in NODE-3285, I've captured the suggestion there, thanks!